modules/gapi/src/api/gmat.cpp

```diff
@@ -36,26 +36,34 @@ const cv::GOrigin& cv::GMat::priv() const
     return *m_priv;
 }
 
-std::tuple<int,int> cv::gapi::detail::checkVector(const cv::GMatDesc& in, const int n,
-                                                  const int expectedDepth)
+bool cv::gapi::detail::checkVector(const cv::GMatDesc& in, const int n, const int expectedDepth,
```

Comment:
may be it is better to have 2 overloads, one which calculates dimensionality and one which checks for expected dimensionality

You just need to write two overloads which will call inside your function:

`bool cv::gapi::detail::checkVector(const cv::GMatDesc& in, const int expectedDepth, int& quantity, int& outDim)`
`bool cv::gapi::detail::checkVector(const cv::GMatDesc& in, const int inDim, const int expectedDepth, int& quantity)`

Both of them will call this func and this func will be named as checkVectorInternal or smth similar

If this costs a lot, may be we need to create a JIRA for that

Reply:
May be 3 overloads, where `quantity` out argument can be also omitted

Reply:
I thought about that. Actually, sometimes one doesn't need even to calculate `quantity`, so perfect will be to implement 3 overloads.
But is it really necessary?

Also, if we speak about two overloads, I'd rather split the core of the function to two as there is actually mix of two functionalities

On the whole, it is rather difficult to decide, this needs more precise discussion; so I suggest to move this fix to other PR later.
Will add FIXME

Reply:
Changed, separated two functions